libraries/AP_SerialManager/AP_SerialManager.cpp

```diff
@@ -98,26 +124,26 @@ extern const AP_HAL::HAL& hal;
   supported serial device type has it's own parameter within AP_Periph
   for which port is used.
  */
-#undef SERIAL0_PROTOCOL
-#undef SERIAL1_PROTOCOL
-#undef SERIAL2_PROTOCOL
-#undef SERIAL3_PROTOCOL
-#undef SERIAL4_PROTOCOL
-#undef SERIAL5_PROTOCOL
-#undef SERIAL6_PROTOCOL
-#undef SERIAL7_PROTOCOL
-#undef SERIAL8_PROTOCOL
-#undef SERIAL9_PROTOCOL
-#define SERIAL0_PROTOCOL SerialProtocol_None
-#define SERIAL1_PROTOCOL SerialProtocol_None
-#define SERIAL2_PROTOCOL SerialProtocol_None
-#define SERIAL3_PROTOCOL SerialProtocol_None
-#define SERIAL4_PROTOCOL SerialProtocol_None
-#define SERIAL5_PROTOCOL SerialProtocol_None
-#define SERIAL6_PROTOCOL SerialProtocol_None
-#define SERIAL7_PROTOCOL SerialProtocol_None
-#define SERIAL8_PROTOCOL SerialProtocol_None
-#define SERIAL9_PROTOCOL SerialProtocol_None
+#undef DEFAULT_SERIAL0_PROTOCOL
+#undef DEFAULT_SERIAL1_PROTOCOL
+#undef DEFAULT_SERIAL2_PROTOCOL
+#undef DEFAULT_SERIAL3_PROTOCOL
+#undef DEFAULT_SERIAL4_PROTOCOL
+#undef DEFAULT_SERIAL5_PROTOCOL
+#undef DEFAULT_SERIAL6_PROTOCOL
+#undef DEFAULT_SERIAL7_PROTOCOL
+#undef DEFAULT_SERIAL8_PROTOCOL
+#undef DEFAULT_SERIAL9_PROTOCOL
+#define DEFAULT_SERIAL0_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL2_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL3_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL4_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL5_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL6_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL7_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL8_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL9_PROTOCOL SerialProtocol_None
```

Comment:
I'll probably move this crap down into `chibios_hwdef.py` where we set the defaults for AP_Periph